apps/api-gateway/src/issuance/dtos/issuance.dto.ts

```diff
@@ -81,30 +84,155 @@ export class Credential {
+    public constructor(options: CredentialPreviewAttributeOptions) {
+      if (options) {
+        this.name = options.name;
+        this.mimeType = options.mimeType;
+        this.value = options.value;
+      }
+    }
+  
+    @ApiProperty({ example: 'name' })
+    @IsNotEmpty({message: 'name cannot be empty'})
+    @IsString()
+    public name!: string;
+  
+  
+    @ApiProperty({ example: 'text/plain' })
+    @IsOptional()
+    @IsMimeType()
+    public mimeType?: string = 'text/plain';
+  
+    @ApiProperty({ example: 'Alice' })
+    @IsNotEmpty({message: 'value cannot be empty'})
+    @IsString()
+    public value!: string;
+  }
+
+  export interface CredentialPreviewOptions {
+  attributes: CredentialPreviewAttributeOptions[]
+  }
+
+  
+  export class AnoncredsObject {
+  
+    @ApiProperty({type: [CredentialPreviewAttribute]})
+    @ArrayMinSize(1)
+    @ValidateNested()
+    @Type(() => CredentialPreviewAttribute)
+    attributes: CredentialPreviewAttribute[];
+  
+    @ApiProperty({ example: 'Mdst5jN9uavVM8tq4qgxUm:3:CL:60112:default' })
+    @IsNotEmpty({ message: 'Credential definition Id is required' })
+    @IsString({ message: 'Credential definition id should be string' })
+    @Transform(({ value }) => value.trim())
+    credentialDefinitionId: string;
+  }
+  
+  export class AttributesDto {
+    @ApiProperty({ example: 'name' })
+    @IsNotEmpty({message: 'name cannot be empty'})
+    @IsString()
+    public name!: string;
+  
+    @ApiProperty({ example: 'Alice' })
+    @IsNotEmpty({message: 'value cannot be empty'})
+    @IsString()
+    public value!: string;
+  }
+  
+  export class IndyObject {
+    @ApiProperty({ type: [AttributesDto]})
+    @ArrayMinSize(1)
+    @IsNotEmpty()
+    @ValidateNested()
+    @Type(() => AttributesDto)
+    attributes: AttributesDto[];
+  
+    @ApiProperty({ example: 'Mdst5jN9uavVM8tq4qgxUm:3:CL:60112:default' })
+    @IsNotEmpty({ message: 'Credential definition Id is required' })
+    @IsString({ message: 'Credential definition id should be string' })
+    @Transform(({ value }) => value.trim())
+    credentialDefinitionId: string;
+  }
+  
+  export class JsonLdObject {
+    @ApiProperty({ type: Credential })
+    @IsNotEmpty({ message: 'Please provide valid credential' })
+    @ValidateNested()
+    @Type(() => Credential)
+    credential: Credential;
+  
+    @ApiProperty({ type: JsonLdCredentialDetailOptions })
+    @IsNotEmpty({ message: 'Please provide valid options' })
+    @ValidateNested()
+    @Type(() => JsonLdCredentialDetailOptions)
+    options: JsonLdCredentialDetailOptions;
+  }
+  
+  export class IndyDto {
+    @ApiProperty({type: IndyObject})
+    @IsNotEmpty()
+    @ValidateNested()
+    @Type(() => IndyObject)
+    indy: IndyObject;
+  }
+  
+  export class AnonCredsDto {
+    @ApiProperty({type: AnoncredsObject})
+    @IsNotEmpty()
+    @ValidateNested()
+    @Type(() => AnoncredsObject)
+    anoncreds: AnoncredsObject;
+  }
+  
+  export class JsonLdDto {
```

Comment:
```suggestion
  export class JsonldDto {
```

Reply:
Here, JsonLd is JSON for Linking Data hence we have used camle case to specify that